### PR TITLE
Use native Ruby implementations in Yast::IP (only)

### PR DESCRIPTION
**Pull request by mvidner**
This is a restriction of #310 to only the IP class (excluding URL which still has unresolved issues).

### CODE REVIEW

**Comment by mvidner** [comment]
LGTM, but I've been looking at this for too long, somebody else should check.

File: library/types/src/modules/IP.rb [hunk 1]
```diff
@@ -272,32 +241,23 @@ def CheckNetworkShared(network)
...
       generic_check = CheckNetworkShared(network)
       if generic_check != nil
-        return generic_check 
+        return generic_check
 
-        # 192.168.0.1, 0.8.55.999
-      elsif Check4(network)
-        return true 
+      # 192.168.0.0/20, 0.8.55/158
+      elsif network =~ Regexp.new("^[" + @ValidChars4 + "]+/[0-9]+$")
```
Comment (jreidinger) [inline library/types/src/modules/IP.rb]: it always make me crazy why ValidChars are not constant, then also this regexp can be constant and many others, making code easier to read

**Comment by jreidinger** [comment]
other then small idea to improve ValidChars it looks good for me

File: library/types/src/modules/IP.rb [hunk 2]
```diff
@@ -151,34 +141,25 @@ def ValidNetwork
...
+      IPAddr.new(ip, Socket::AF_INET).to_s
     end
 
     # Converts IPv4 address from string to hex format
     # @param [String] ip IPv4 address as string in "ipv4" format
     # @return [String] representing IP in Hex
-    # @example IP::ToHex("192.168.1.1") -> "0xC0A80101"
-    # @example IP::ToHex("10.10.0.1") -> "0x0A0A0001"
+    # @example IP::ToHex("192.168.1.1") -> "C0A80101"
+    # @example IP::ToHex("10.10.0.1") -> "0A0A0001"
```
Comment (ancorgs) [inline library/types/src/modules/IP.rb]: Does this mean that the behavior of the class actually changed?

Comment (mvidner) [inline library/types/src/modules/IP.rb]: It hasn't, as you can see in the unchanged tests: https://github.com/yast/yast-yast2/blob/573553179d769ac5dcc08d37c05baeabfbc921c9/library/types/test/ip_test.rb#L146
Just the docs was wrong.

File: library/types/src/modules/IP.rb (hunk 1 above)
Comment (mvidner) [inline library/types/src/modules/IP.rb]: Code in other modules uses the variables, so we must keep that API. I will leave it as it is now.

Comment (jreidinger) [inline library/types/src/modules/IP.rb]: but only read-only if I remember correctly, so we can make accessor to constant. It is confusing to have editable ValidChars

**Pull request merged**